docs/userguides/plugins.md

```diff
@@ -1,120 +1,142 @@
-used to compile files, contract types which come from the compiled files
-that you might want to use in the package.
-[ContractType](autoapi/ape/types/contract/index.html#ape.types.contract.ContractType)
-
-### Account registration process flow
-
-accounts manager AccountAPI/AccountContainerAPI plugins Signing messages
-and transactions via AccountAPI
-
-### Transactional process flow
-
-networks manager EcosystemAPI plugins NetworkAPI plugins ProviderAPI
-plugins ExplorerAPI plugins ContractInstance type and encoding via
-EcosystemAPI
-
-### Argument conversion process flow
-
-CLI arguments are decoded and passed in to the application with `click`.
-
-## Writing CLI plugins
-
-CLI plugins will use the plugin registration process defined above. The
-CLI plugins should use the `click` library in order to be able to supply
-arguments from the CLI.
+# Developing Plugins
+
+Your plugin project can be any type of python project. To create an `ape` plugin, implement one or more API classes 
+from the the `ape.api` namespace or add key `ape_cli_subcommands` to your entry-points list in your project's 
+`setup.py`, depending on what type of plugin you want to create. This guide is intended to assist in both of those 
+use-cases.
+
+The following is a list of example plugins to use as a reference when developing plugins:
+
+* `CompilerAPI`: [the Solidity plugin](https://github.com/apeworx/ape-solidity)
+* `ProviderAPI`: [the Infura plugin](https://github.com/apeworx/ape-infura)
+* `AccountAPI`: [the Trezor plugin](https://github.com/apeworx/ape-trezor)
+* `CLI Extension`: [the Tokens-List plugin](https://github.com/apeworx/ape-tokens)
+
+## Initialize a Plugin Project
+
+As previously mentioned, a plugin project is merely a python project. However, you can optionally use this 
+[project template](https://github.com/ApeWorX/project-template) for initializing your plugin. **NOTE**: this template 
+is primarily designed for plugins built within the ApeWorX team organization; not everything may apply. It is okay to 
+delete anything that does not work or that you don't find helpful. The template may be good to follow if you want to 
+keep your plugin of similar quality to plugins developed by the ApeWorX team.
+
+## Implementing API Classes
+
+API classes (classes from the `ape.api` namespace) are primary composed of abstract methods and properties that plugins 
+must implement. A benefit of the plugin system is that each plugin can implement these however they need, so long as 
+they conform to the API interface. Two plugins with the same API may do entirely different things and yet be 
+interchangeable in their usage.
+
+To implement an API, import its class and use it as a base-class in your implementation class. **WARNING**: The plugin 
+will fail to work properly if you do not implement all the abstract methods.
+
+```python
+from ape.api import ProviderAPI
+
+
+class MyProvider(ProviderAPI):
+  """Implement the abstracts methods here."""
+```
```

Comment:
Not necessary, but maybe show one abstract method in the `ProviderAPI` being overridden?